src/schema/v2/artwork/updateArtworkMutation.ts

```diff
@@ -69,20 +111,44 @@ export const updateArtworkMutation = mutationWithClientMutationId<
     },
   },
   mutateAndGetPayload: async (
-    { availability, id },
-    { updateArtworkLoader }
+    { id, editionSets, ...args },
+    { updateArtworkLoader, updateArtworkEditionSetLoader }
   ) => {
-    if (!updateArtworkLoader) {
+    if (!updateArtworkLoader || !updateArtworkEditionSetLoader) {
       return new Error("You need to be signed in to perform this action")
     }
 
+    const getGravityArgs = (inputArgs: UpdateArtworkMutationInputProps) => {
+      return {
+        id: inputArgs.id,
+        availability: inputArgs.availability,
+        ecommerce: inputArgs.ecommerce,
+        display_price_range: inputArgs.displayPriceRange,
+        offer: inputArgs.offer,
+        price_hidden: inputArgs.priceHidden,
+        price_listed: inputArgs.priceListed,
+      }
+    }
+
     try {
-      const response = await updateArtworkLoader(id, {
-        availability,
-      })
+      if (editionSets?.length) {
+        await Promise.all(
```

Comment:
I guess this makes sense - maybe Gravity should have an endpoint that can just accept this entire bag of variables from the mutation and do the right thing - versus a bit of this necessary logic needing to live here (but also totally fine).

Reply:
I suspect this will change, but this loader was already here so why not